central/sensor/service/connection/ratelimit_manager_test.go

```diff
@@ -0,0 +1,186 @@
+	assert.Len(t, m.initSyncSensors, 1)
+
+	m.removeCluster("test-1")
+	assert.Len(t, m.initSyncSensors, 0)
+}
+
+func TestNewRateLimitManagerNegativeMaxInitSync(t *testing.T) {
+	t.Setenv(env.CentralMaxInitSyncSensors.EnvVar(), "-1")
+	assert.Panics(t, func() { newRateLimitManager() })
+}
+
+func TestNewRateLimitManagerZeroMaxInitSync(t *testing.T) {
+	t.Setenv(env.CentralMaxInitSyncSensors.EnvVar(), "0")
+	m := newRateLimitManager()
+	assert.Equal(t, math.MaxInt, m.initSyncMaxSensors)
+
+	assert.True(t, m.addInitSync("test-1"), "Can add if limit is set to 0")
+	assert.Len(t, m.initSyncSensors, 1)
+
+	m.removeCluster("test-1")
+	assert.Len(t, m.initSyncSensors, 0)
+}
+
+func TestNewRateLimitManagerMaxInitSync(t *testing.T) {
+	centralMaxInitSyncSensors := 5
+
+	t.Setenv(env.CentralMaxInitSyncSensors.EnvVar(), fmt.Sprintf("%d", centralMaxInitSyncSensors))
+	m := newRateLimitManager()
+
+	for i := 0; i < centralMaxInitSyncSensors; i++ {
+		assert.True(t, m.addInitSync(fmt.Sprintf("test-%d", i)))
+	}
+	assert.False(t, m.addInitSync("test-a"), "Unable to add after limit is reached")
+	assert.Len(t, m.initSyncSensors, centralMaxInitSyncSensors)
+
+	m.removeCluster("test-a")
+	assert.False(t, m.addInitSync("test-a"), "Unable to add after removing non-existing")
+
+	m.removeCluster("test-1")
+	assert.Len(t, m.initSyncSensors, centralMaxInitSyncSensors-1)
+	assert.True(t, m.addInitSync("test-a"), "Can add after one is removed")
+	assert.Len(t, m.initSyncSensors, centralMaxInitSyncSensors)
+
+	assert.False(t, m.addInitSync("test-b"), "Unable to add after limit is reached")
+}
+
+func TestRateLimitManagerNilGuards(t *testing.T) {
+	tests := []struct {
+		name    string
+		manager *rateLimitManager
+	}{
+		{"nil manager", nil},
+		{"empty manager", &rateLimitManager{initSyncMaxSensors: 1}},
+		{"no tracker", &rateLimitManager{initSyncMaxSensors: 1, msgRateLimiter: ratelimit.NewRateLimiter(10, time.Minute)}},
+		{"no rate limiter", &rateLimitManager{initSyncMaxSensors: 1, msgRateTracker: ratetracker.NewClusterRateTracker(time.Minute, 1)}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.True(t, tt.manager.addInitSync("test-1"))
+			assert.False(t, tt.manager.limitClusterMsg("test-1"))
+			assert.False(t, tt.manager.throttleMsg("test-1"))
+			assert.NotPanics(t, func() { tt.manager.removeCluster("test-1") })
+		})
+	}
+}
+
+func TestNewRateLimitManagerEventsPerSecond(t *testing.T) {
+	t.Setenv(env.CentralSensorMaxEventsPerSecond.EnvVar(), "3")
+	t.Setenv(env.CentralSensorMaxEventsThrottleDuration.EnvVar(), "0")
+	m := newRateLimitManager()
+
+	hitLimit := false
+	for i := 0; i < 30; i++ {
+		limitMsg := m.limitClusterMsg("c1")
+		if i < 3 {
+			require.False(t, limitMsg, "Limit is not reached")
+			continue
+		}
+
+		if limitMsg {
+			hitLimit = true
+			break
+		}
+	}
+	assert.True(t, hitLimit)
+
+	// Wait for rate limit to refill.
+	time.Sleep(time.Second)
```

Comment:
I'm not a fan of sleeping in tests. If we need to test it then maybe allow mocking rate limiter to get full control what's happening.